bridges/YoutubeBridge.php

```diff
@@ -221,31 +202,164 @@ private function ytGetSimpleHTMLDOM($url, $cached = false){
 			$defaultSpanText);
 	}
 
+	private function getJSONData($html) {
+		$scriptRegex = '/var ytInitialData = (.*?);<\/script>/';
+		preg_match($scriptRegex, $html, $matches) or returnServerError('Could not find ytInitialData');
+		return json_decode($matches[1]);
+	}
+
+	private function parseJSONListing($jsonData) {
+		$duration_min = $this->getInput('duration_min') ?: -1;
+		$duration_min = $duration_min * 60;
+
+		$duration_max = $this->getInput('duration_max') ?: INF;
+		$duration_max = $duration_max * 60;
+
+		if($duration_max < $duration_min) {
+			returnClientError('Max duration must be greater than min duration!');
+		}
+
+		$count = 0;
+		foreach($jsonData as $item) {
+			$count++;
+			if($count == 20) {
+				break;
+			}
```

Comment:
Playlist can have videos sorted by ascending (oldest first) upload date. This construction will ignore latest videos, which is not good.

For example: https://www.youtube.com/playlist?list=PLaDrN74SfdT7Ueqtwn_bXo1MuSWT0ji2w